lib/order.rb

```diff
@@ -1,18 +1,241 @@
+        print "TOTAL: "
+        puts index.total
+        n += 1
+      end
+    end
+
+    def self.find(id)
+      if
+        @all_orders.include?(@all_orders[id - 1])
+        @all_orders[id - 1].receipt
+        puts "Total: #{@all_orders[id - 1].total}"
+      else
+        puts "That order does not exist."
+      end
+    end
+
+
+    def receipt
+      puts "RECEIPT:"
+      @item_and_price_hash.each do |key, value|
+        puts "#{key}: #{value}"
+      end
     end
 
+
     def total
-      # TODO: implement total
+      if
+        @products.empty?
+        return 0
+      end
+
+      total = 0
+      tax = 0.075
+      @item_and_price_hash.each do |item, price|
+        total += price.to_f
+      end
+      total = total + (total * tax)
+      return total.round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      @product_name = product_name
+      @product_price = product_price.to_f
+      if @item_and_price_hash.has_key? @product_name
+        return false
+      else
+        @item_and_price_hash[@product_name] = @product_price
+        return @item_and_price_hash
+      end
+    end
+
+    def remove_product(product_name)
+      if @item_and_price_hash.has_key?(product_name)
+        @item_and_price_hash.delete(product_name)
+      else
+        puts "Product is not on this order."
+      end
+    end
+  end
+
+  class OnlineOrder < Order
+    attr_reader :customer_id
+    attr_accessor :status
+
+    def initialize(id, products, customer_id, status = :pending)
+      super(id, products)
+      @customer_id = customer_id.to_i
+      @status = status
+      if
+        @status == nil
+        @status = :pending
+      else
+        @status = status.to_sym
+      end
+    end
+
+    def self.all(data_collection)
+      @all_orders = []
+      data_collection.each do |row|
+        order = Grocery::OnlineOrder.new(row[0], row[1], row[2], row[3])
+        @all_orders << order
+      end
+      n = 1
+      @all_orders.each do |index|
+        puts "\n\nORDER #{n}:"
+        index.receipt
+        print "TOTAL: "
+        puts index.total
+        print "CUSTOMER ID: "
+        puts index.customer_id
+        print "STATUS: "
+        puts index.status
+        n += 1
+      end
+    end
+
+    def self.find(id)
+       puts"\nSTATUS: #{@all_orders[id - 1].status}"
+       puts"CUSTOMER ID: #{@all_orders[id - 1].customer_id}"
+      return super
+    end
+
+    def self.find_by(customer_id)
+      puts "ORDERS BY CUSTOMER ##{customer_id}:"
+      @all_orders.each do |index|
+        if index.customer_id == customer_id
+          puts index.id
+        end
+      end
+    end
+
+    def total
+      puts "\n*$10 SHIPPING FEE HAS BEEN ADDED*"
+      return super + 10
+    end
+
+    def add_product(product_name, product_price)
+      @product_name = product_name
+      @product_price = product_price.to_f
+      if
+        @item_and_price_hash.has_key? @product_name
```

Comment:
I'd like to work on this habit that I think you're forming about `if` conditionals: `if` statements have an expression that they evaluate next to them. In this case, it's `if @item_and_price_hash.has_key? @product_name`.

In this project, pretty often you put the `if` and the `expression` on different lines: even though it works, it's out of convention and it makes it a little difficult to read. It'd be great if these were on the same line from now on